groups/stack/main.tf

```diff
@@ -0,0 +1,193 @@
+provider "aws" {
+  region  = var.aws_region
+  version = "~> 2.32.0"
+}
+
+terraform {
+  backend "s3" {
+  }
+}
+
+# Configure the remote state data source to acquire configuration
+# created through the code in ch-service-terraform/aws-mm-networks.
+data "terraform_remote_state" "networks" {
+  backend = "s3"
+  config = {
+    bucket = var.remote_state_bucket
+    key    = "${var.state_prefix}/${var.deploy_to}/${var.deploy_to}.tfstate"
+    region = var.aws_region
+  }
+}
+locals {
+  vpc_id            = data.terraform_remote_state.networks.outputs.vpc_id
+  application_ids   = data.terraform_remote_state.networks.outputs.application_ids
+  application_cidrs = data.terraform_remote_state.networks.outputs.application_cidrs
+  public_ids        = data.terraform_remote_state.networks.outputs.public_ids
+  public_cidrs      = data.terraform_remote_state.networks.outputs.public_cidrs
+}
+
+# Configure the remote state data source to acquire configuration created
+# through the code in aws-common-infrastructure-terraform/groups/networking.
+data "terraform_remote_state" "networks_common_infra" {
+  backend = "s3"
+  config = {
+    bucket = var.aws_bucket
+    key    = "aws-common-infrastructure-terraform/common-${var.aws_region}/networking.tfstate"
+    region = var.aws_region
+  }
+}
+locals {
+  internal_cidrs = values(data.terraform_remote_state.networks_common_infra.outputs.internal_cidrs)
+  vpn_cidrs      = values(data.terraform_remote_state.networks_common_infra.outputs.vpn_cidrs)
+}
+
+# Remote state data source for Ireland, required for Concourse management CIDRs
+data "terraform_remote_state" "networks_common_infra_ireland" {
+  backend = "s3"
+  config = {
+    bucket = "development-eu-west-1.terraform-state.ch.gov.uk"
+    key    = "aws-common-infrastructure-terraform/common-eu-west-1/networking.tfstate"
+    region = "eu-west-1"
+  }
+}
+locals {
+  management_private_subnet_cidrs = values(data.terraform_remote_state.networks_common_infra_ireland.outputs.management_private_subnet_cidrs)
+}
+
+# Configure the remote state data source to acquire configuration
+# created through the code in the services-stack-configs stack in the
+# aws-common-infrastructure-terraform repo.
+data "terraform_remote_state" "services-stack-configs" {
+  backend = "s3"
+  config = {
+    bucket = var.aws_bucket # aws-common-infrastructure-terraform repo uses the same remote state bucket
+    key    = "aws-common-infrastructure-terraform/common-${var.aws_region}/services-stack-configs.tfstate"
+    region = var.aws_region
+  }
+}
```

Comment:
Remote state data sources should be avoided, where possible, in faviour of obtaining data from Vault or via other data source methods.